packages/@apphosting/adapter-nextjs/src/utils.ts

```diff
@@ -7,15 +7,17 @@ import { stringify as yamlStringify } from "yaml";
 import { spawnSync } from "child_process";
 
 import { join, relative, normalize } from "path";
+import { createRequire } from "node:module";
 
 import type { RoutesManifest } from "./interfaces.js";
 // fs-extra is CJS, readJson can't be imported using shorthand
 export const { move, exists, writeFile, readJson } = fsExtra;
 
 export async function loadConfig(cwd: string) {
+  const require = createRequire(import.meta.url);
   // dynamically load NextJS so this can be used in an NPX context
   const { default: nextServerConfig }: { default: typeof import("next/dist/server/config.js") } =
-    await import(`${cwd}/node_modules/next/dist/server/config.js`);
+    await import(require.resolve("next/dist/server/config.js", { paths: [cwd] }));
```

Comment:
I have a pr out for fixing yarn pnp in nextjs and it includes this change + a little extra to get yarn pnp working, https://github.com/FirebaseExtended/firebase-framework-tools/pull/156/files.

Could you revert these changes and we work off of that pr going forward?

Reply:
Actually lets get your changes in instead of my new pr. I think we should get pnpm working and maybe try to get everything setup with corepack as a longer term solution

Reply:
ok sounds good, let me retest and submit this one. Will take a look at corepack later